server/stream.go

```diff
@@ -455,18 +461,35 @@ func (a *Account) addStreamWithAssignment(config *StreamConfig, fsConfig *FileSt
 	}
 
 	// Setup our internal indexed names here for sources and check if the transform (if any) is valid.
-
 	for _, ssi := range cfg.Sources {
-		// check the filter, if any, is valid
-		if ssi.FilterSubject != _EMPTY_ && !IsValidSubject(ssi.FilterSubject) {
-			jsa.mu.Unlock()
-			return nil, fmt.Errorf("subject filter '%s' for the source %w", ssi.FilterSubject, ErrBadSubject)
-		}
-		// check the transform, if any, is valid
-		if ssi.SubjectTransformDest != _EMPTY_ {
-			if _, err = NewSubjectTransform(ssi.FilterSubject, ssi.SubjectTransformDest); err != nil {
+		if len(ssi.FilterSubjects) == 0 {
+			// check the filter, if any, is valid
+			if ssi.FilterSubject != _EMPTY_ && !IsValidSubject(ssi.FilterSubject) {
 				jsa.mu.Unlock()
-				return nil, fmt.Errorf("subject transform from '%s' to '%s' for the source %w", ssi.FilterSubject, ssi.SubjectTransformDest, err)
+				return nil, fmt.Errorf("subject filter '%s' for the source %w", ssi.FilterSubject, ErrBadSubject)
+			}
+			// check the transform, if any, is valid
+			if ssi.SubjectTransformDest != _EMPTY_ {
+				if _, err = NewSubjectTransform(ssi.FilterSubject, ssi.SubjectTransformDest); err != nil {
+					jsa.mu.Unlock()
+					return nil, fmt.Errorf("subject transform from '%s' to '%s' for the source %w", ssi.FilterSubject, ssi.SubjectTransformDest, err)
+				}
+			}
+		} else {
+			for i, sf := range ssi.FilterSubjects {
+				if sf != _EMPTY_ && !IsValidSubject(sf) {
+					jsa.mu.Unlock()
+					return nil, fmt.Errorf("subject filter '%s' for the source %w", sf, ErrBadSubject)
+				}
+				// check the transform, if any, is valid
+				if len(ssi.SubjectTransformDests) != 0 {
```

Comment:
Usage of two slices, one for filters, and one for destinations has in my opinion few cons:
* **User has to specify destinations for all or no filters. If this was slice of Structs, then they could specify destinations for some, and no destination for others. Wouldn't that be useful?**
* Having two slices requires manual checks of len of them. That's a runtime check that is easy to get wrong. I know when have it in other place in the PR, but is that path always run before this loop? Can future unrelated PRs change it, causing panic out of range?
* It's pretty unintuitive for users to specify two aligned separate slices, that are related.  I think clients will anyway expose `[]struct{subject, Option<destination>}` (pseudo code). Will discuss it today on the call the have final agreement on client side API.